### PR TITLE
Add APIv4 endpoint to permanently delete teams

### PR DESCRIPTION
**Pull request by ZJvandeWeg**
#### Summary

Add APIv4 endpoint to permanently delete teams

#### Ticket Link

None to my knowledge

#### Checklist

- [X] Added or updated unit tests (required for all new features)
- [x] Added API documentation (required for all new APIs)
- [X] All new/modified APIs include changes to the drivers
- [ ] Tests are passing

### CODE REVIEW

File: config/config.json [hunk 1]
```diff
@@ -47,20 +47,12 @@
         "EnableUserStatuses": true,
         "ClusterLogTimeoutMilliseconds": 2000
     },
-    "ElasticSearchSettings": {
```
Comment (ZJvandeWeg) [inline config/config.json]: Ill checkout master after the first review to fix stuff in one go.

**Comment by jwilander** [comment]
Hi @ZJvandeWeg, we're just going through our release process right now but I'll take a look in the next couple days

**Comment by ZJvandeWeg** [comment]
@jwilander Well, there is a test failure which is legit 😉 

But I wonder what you mean with 

>  minor spelling mistake

I don't see a diff note on that?

File: model/client4.go [hunk 2]
```diff
@@ -1046,6 +1046,17 @@ func (c *Client4) SoftDeleteTeam(teamId string) (bool, *Response) {
 	}
 }
 
+// PermanentDeleteTeam deletes the team, should only be used when needed for
+// complience and the like
```
Comment (jwilander) [inline model/client4.go]: Minor typo, `complience` -> `compliance`

**Comment by jwilander** [comment]
Sorry, forgot to submit the comment. And yes, the test failure needs to be fixed but the code looks good from inspection

**Comment by ZJvandeWeg** [comment]
@jwilander Amended my fixes, the test now only checks if the soft deletion property is set as the rest happens in the background and seemed the only property that can be tested. Let me know what you think about it.

**Comment by jwilander** [comment]
I think that's fine for now. We don't have a good mechanism yet for verifying results of code that runs in unbounded goroutines and I'd rather avoid using sleeps.

**Pull request merged**